docs/name-owners/atomic-name-trading/index.md

```diff
@@ -0,0 +1,294 @@
+
+```
+$ namecoind name_show d/my-cool-domain
+{
+  "name" : "d/my-cool-domain",
+  "value" : "for sale by Sally for 10 NMC, pay to mm1mGfLEMjFPyGsbkcoYiN6yiPcBztK6Dw",
+  "txid" : "00abd8282558574369038722d24c65355e1b98ca3f00d7d09cb7bea8732b598a",
+  "address" : "mt2vTvKfvfb9TKfxB8ns7NVrY4cYzRs73b",
+  "expires_in" : 36000
+}
+```
+
+Now, Bob can prepare an atomic transaction that sends 10 NMC to Sally and transfers __d/my-cool-domain__ at the same time to himself. We assume that Sally wants the 10 NMC sent to __mm1mGfLEMjFPyGsbkcoYiN6yiPcBztK6Dw__, while Bob wants the name to be transferred to __mwKDtLZ9EHSeL8TjkZAcppsnb7M3LRhoGR__. 
+
+To construct the transaction, Bob needs to use the "createrawtransaction" RPC command. It needs three pieces of information: 
+
+* Currency inputs: These will be provided by Bob to pay the 10 NMC to Sally as well as the transaction fee.
+* Currency outputs: 10 NMC to Sally's address and potential change back to Bob.
+* A name_update operation to be performed.
+
+In this example, these three parts will be written in seperate files for simplicity, and then these files will be used in the final RPC command. Of course, it is also possible to write everything directly onto the command-line. 
+
+### Currency Inputs
+
+Assuming a transaction fee of 0.005 NMC is appropriate, Bob needs to find one of his unspent inputs of at least 10.005 NMC to include in the transaction. For this, the "listunspent" command can be used. A suitable input could be this one: 
+
+```
+$ namecoind listunspent
+[
+  {
+    "txid" : "12ed9f83b83e248e2ca19e0197933914013145981fcacc5ba47c27e60eacbfaa",
+    "vout" : 0,
+    "scriptPubKey" : "41040a1b1f8d3b497c7ee13e441539b9e1fe5b4ea96696a4827a74e02aa6c4df4383209de42bbc02ce90e28e5e109b9b1b7dcfd14b02449603c4320faf394719a8f1ac",
+    "amount" : 974.90000000,
+    "confirmations" : 2
+  }
+]
+```
+
+With a size of 974.9 NMC, it is large enough. It will produce __974.9 - 10.005 = 964.895 NMC__ change. The interesting data fields for our purpose are __txid__ and __vout__. They will form the currency input for "createrawtransaction", so Bob writes into the file "tx.in": 
+
+
+```
+[{
+  "txid": "12ed9f83b83e248e2ca19e0197933914013145981fcacc5ba47c27e60eacbfaa",
+  "vout": 0
+}]
+```
+
+If no single unspent output is large enough, Bob could instead either include multiple inputs into this file, or produce a suitable output first by sending a sufficiently large amount to himself.
+
+### Currency Outputs
+
+Required currency outputs are the price paid to Sally as well as the change paid back to Bob. So Bob first creates a new address of his own to receive the change. We will use __mwAEUoA3NzSmASyWxrTK1mhsd4pFvruoXj__ for this purpose. Then, the corresponding data to be put in "tx.out" is: 
+
+```
+{
+  "mm1mGfLEMjFPyGsbkcoYiN6yiPcBztK6Dw": 10.0,
+  "mwAEUoA3NzSmASyWxrTK1mhsd4pFvruoXj": 964.895
+}
+```
+
+### The Name Operation
```

Comment:
This part of the instructions is outdated.  Instead, the name should first be added as ordinary input/output in the previous steps, and then `namerawtransaction` applied in addition.

Reply:
Any chance you could submit a follow-up PR for this?

Reply:
Ok sure.  If you prefer to just migrate the existing content in this PR, and then work on fixing it (or the non-trivial parts of it) in a second step (not implying you have to do that second step), that's fine.